util/chunk/compare.go

```diff
@@ -217,11 +218,24 @@ func compare(row Row, colIdx int, ad *types.Datum) int {
 // returns the smallest index i such that the value at row i is not less than `ad`.
 func (c *Chunk) LowerBound(colIdx int, ad *types.Datum) (index int, match bool) {
 	index = sort.Search(c.NumRows(), func(i int) bool {
-		cmp := compare(c.GetRow(i), colIdx, ad)
+		cmp := Compare(c.GetRow(i), colIdx, ad)
 		if cmp == 0 {
 			match = true
 		}
 		return cmp >= 0
 	})
 	return
 }
+
+// UpperBound searches on the non-decreasing column colIdx,
+// returns the smallest index i such that the value at row i is larger than `ad`.
+func (c *Chunk) UpperBound(colIdx int, ad *types.Datum) (index int, match bool) {
```

Comment:
The match doesn't make much sense if the value at the index is larger.
What does the name `ad` means?

Reply:
The match is used to indicate if there are any value equal to the datum.